### ISSUE TITLE
close - IO is still pending on closed socket. read:1, write:0, io:0000023DE45AF900

### ISSUE THREAD

**Issue by hsinyinfu** [PowerShell/Win32-OpenSSH#1899]
**"OpenSSH for Windows" version**
1. `OpenSSH_for_Windows_8.1p1, LibreSSL 3.0.2`
2. `OpenSSH_for_Windows_8.6p1, LibreSSL 3.3.3`
 
**Server OperatingSystem**
Ubuntu 18.04.5 LTS

**Client OperatingSystem**
Windows 10 Enterprise

**What is failing**
In my company, I have two PCs in the company network given by IT.
I usually open a CMD window and connect to remote Linux server with command `ssh <account>@<server>`.
However, I found that on my PC1, after disconnecting from remote Linux server, the ssh always shows the following error message at the last line:
`close - IO is still pending on closed socket. read:1, write:0, io:0000023DE45AF900`
But on my PC2, it never does that after disconnecting from the same remote Linux server.

I also tried to connect to other company remote Linux servers from the two PCs, but the results were the same as above.
So it seems that the problem is not on the remote side, but on the local side.

Besides, I tried different ssh toolchain versions.
The Windows 10 built-in ssh client version is `OpenSSH_for_Windows_8.1p1, LibreSSL 3.0.2`.
I downloaded the latest ssh toolchain version `OpenSSH_for_Windows_8.6p1, LibreSSL 3.3.3`, unzip it and set up PATH env variable, and tried again on my PC1.
Both versions shows the same error message after disconnection.

I would like to know how to fix this issue.

**Comment by hsinyinfu** [comment]
Here is the log on my PC1 with ssh version `OpenSSH_for_Windows_8.6p1, LibreSSL 3.3.3`
```
Microsoft Windows [版本 10.0.19042.985]
(c) Microsoft Corporation. 著作權所有，並保留一切權利。
C:\Users\mtk25836>ssh -V
OpenSSH_for_Windows_8.6p1, LibreSSL 3.3.3

                                                                                                                                                                                                     
C:\Users\mtk25836>where ssh
C:\Users\mtk25836\OpenSSH-Win64\ssh.exe
C:\Windows\System32\OpenSSH\ssh.exe


C:\Users\mtk25836>ssh -v mtk25836@mtkslt236 exit                                                                        
OpenSSH_for_Windows_8.6p1, LibreSSL 3.3.3                                                                               
debug1: Reading configuration data C:\\Users\\mtk25836/.ssh/config                                                      
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling                                              
debug1: Connecting to mtkslt236 [10.21.15.164] port 22.                                                                 
debug1: Connection established.                                                                                         
debug1: identity file C:\\Users\\mtk25836/.ssh/id_rsa type 0                                                            
debug1: identity file C:\\Users\\mtk25836/.ssh/id_rsa-cert type -1                                                      
debug1: identity file C:\\Users\\mtk25836/.ssh/id_dsa type -1                                                           
debug1: identity file C:\\Users\\mtk25836/.ssh/id_dsa-cert type -1                                                      
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa type -1                                                         
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa-cert type -1                                                    
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa_sk type -1                                                      
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa_sk-cert type -1                                                 
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519 type -1                                                       
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519-cert type -1                                                  
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519_sk type -1                                                    
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519_sk-cert type -1                                               
debug1: identity file C:\\Users\\mtk25836/.ssh/id_xmss type -1                                                          
debug1: identity file C:\\Users\\mtk25836/.ssh/id_xmss-cert type -1                                                     
debug1: Local version string SSH-2.0-OpenSSH_for_Windows_8.6                                                            
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.6p1 Ubuntu-4ubuntu0.3                            
debug1: compat_banner: match: OpenSSH_7.6p1 Ubuntu-4ubuntu0.3 pat OpenSSH_7.0*,OpenSSH_7.1*,OpenSSH_7.2*,OpenSSH_7.3*,OpenSSH_7.4*,OpenSSH_7.5*,OpenSSH_7.6*,OpenSSH_7.7* compat 0x04000002                                                     
debug1: Authenticating to mtkslt236:22 as 'mtk25836'                                                                    
debug1: load_hostkeys: fopen C:\\Users\\mtk25836/.ssh/known_hosts2: No such file or directory                           
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts: No such file or directory                            
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts2: No such file or directory                           
debug1: SSH2_MSG_KEXINIT sent                                                                                           
debug1: SSH2_MSG_KEXINIT received                                                                                       
debug1: kex: algorithm: curve25519-sha256                                                                               
debug1: kex: host key algorithm: ssh-ed25519                                                                            
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none                     
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none                     
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY                                                                               
debug1: SSH2_MSG_KEX_ECDH_REPLY received                                                                                
debug1: Server host key: ssh-ed25519 SHA256:lCbAZEo9ZlEwDts4qu6eVmdfdBDcQTE02xA46HPsILY                                 
debug1: load_hostkeys: fopen C:\\Users\\mtk25836/.ssh/known_hosts2: No such file or directory                           
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts: No such file or directory                            
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts2: No such file or directory                           
debug1: Host 'mtkslt236' is known and matches the ED25519 host key.                                                     
debug1: Found key in C:\\Users\\mtk25836/.ssh/known_hosts:5                                                             
debug1: rekey out after 134217728 blocks                                                                                
debug1: SSH2_MSG_NEWKEYS sent                                                                                           
debug1: expecting SSH2_MSG_NEWKEYS                                                                                      
debug1: SSH2_MSG_NEWKEYS received                                                                                       
debug1: rekey in after 134217728 blocks                                                                                 
debug1: pubkey_prepare: ssh_get_authentication_socket: No such file or directory                                        
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_rsa RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs        
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_dsa                                                               
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ecdsa                                                             
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ecdsa_sk                                                          
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ed25519                                                           
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ed25519_sk                                                        
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_xmss                                                              
debug1: SSH2_MSG_EXT_INFO received                                                                                      
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>                                                                                 
debug1: SSH2_MSG_SERVICE_ACCEPT received                                                                                
debug1: Authentications that can continue: publickey,password                                                           
debug1: Next authentication method: publickey                                                                           
debug1: Offering public key: C:\\Users\\mtk25836/.ssh/id_rsa RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs     
debug1: Server accepts key: C:\\Users\\mtk25836/.ssh/id_rsa RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs      
debug1: Authentication succeeded (publickey).                                                                           
Authenticated to mtkslt236 ([10.21.15.164]:22).                  
debug1: channel 0: new [client-session]                                                                                 
debug1: Requesting no-more-sessions@openssh.com                                                                         
debug1: Entering interactive session.                                                                                   
debug1: pledge: filesystem full                                                                                         
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0                                         
debug1: client_input_hostkeys: searching C:\\Users\\mtk25836/.ssh/known_hosts for mtkslt236 / (none)                    
debug1: client_input_hostkeys: searching C:\\Users\\mtk25836/.ssh/known_hosts2 for mtkslt236 / (none)                   
debug1: client_input_hostkeys: hostkeys file C:\\Users\\mtk25836/.ssh/known_hosts2 does not exist                       
debug1: client_input_hostkeys: no new or deprecated keys from server                                                    
debug1: Sending command: exit                                                                                           
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0                                                   
debug1: client_input_channel_req: channel 0 rtype eow@openssh.com reply 0                                               
debug1: channel 0: free: client-session, nchannels 1                                                                    
Transferred: sent 3212, received 2628 bytes, in 0.2 seconds                                                             
Bytes per second: sent 17766.5, received 14536.2                                                                        
debug1: Exit status 0                                                                                                   
close - IO is still pending on closed socket. read:1, write:0, io:0000023DE45AF900
```

**Comment by bagajjal** [comment]
@hsinyinfu - You can compare the output of ssh.exe -vvv user@ip on both the machines. On the problematic machine, the socket is closed while there is some data pending for read.

**Comment by hsinyinfu** [comment]
Here is the log with command `ssh.exe -vvv user@ip' on the problematic machine. Could you please tell me where it shows that some data pending for read?
```
OpenSSH_for_Windows_8.6p1, LibreSSL 3.3.3
debug1: Reading configuration data C:\\Users\\mtk25836/.ssh/config
debug3: Failed to open file:C:/ProgramData/ssh/ssh_config error:2
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts' -> 'C:\\Users\\mtk25836/.ssh/known_hosts'
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts2' -> 'C:\\Users\\mtk25836/.ssh/known_hosts2'
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug2: resolving "mtkslt236" port 22
debug3: ssh_connect_direct: entering
debug1: Connecting to mtkslt236 [10.21.15.164] port 22.
debug1: Connection established.
debug1: identity file C:\\Users\\mtk25836/.ssh/id_rsa type 0
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_rsa-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_rsa-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_rsa-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_rsa-cert type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_dsa error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_dsa.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_dsa error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_dsa type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_dsa-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_dsa-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_dsa-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_dsa-cert type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa-cert type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa_sk error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa_sk.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa_sk error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa_sk type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa_sk-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa_sk-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ecdsa_sk-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ecdsa_sk-cert type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519 error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ed25519 error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519 type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ed25519-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519-cert type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519_sk error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519_sk.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ed25519_sk error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519_sk type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519_sk-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_ed25519_sk-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_ed25519_sk-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_ed25519_sk-cert type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_xmss error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_xmss.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_xmss error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_xmss type -1
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_xmss-cert error:2
debug3: Failed to open file:C:/Users/mtk25836/.ssh/id_xmss-cert.pub error:2
debug3: failed to open file:C:/Users/mtk25836/.ssh/id_xmss-cert error:2
debug1: identity file C:\\Users\\mtk25836/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_for_Windows_8.6
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.6p1 Ubuntu-4ubuntu0.3
debug1: compat_banner: match: OpenSSH_7.6p1 Ubuntu-4ubuntu0.3 pat OpenSSH_7.0*,OpenSSH_7.1*,OpenSSH_7.2*,OpenSSH_7.3*,OpenSSH_7.4*,OpenSSH_7.5*,OpenSSH_7.6*,OpenSSH_7.7* compat 0x04000002
debug2: fd 3 setting O_NONBLOCK
debug1: Authenticating to mtkslt236:22 as 'mtk25836'
debug3: record_hostkey: found key type ECDSA in file C:\\Users\\mtk25836/.ssh/known_hosts:3
debug3: record_hostkey: found key type RSA in file C:\\Users\\mtk25836/.ssh/known_hosts:4
debug3: record_hostkey: found key type ED25519 in file C:\\Users\\mtk25836/.ssh/known_hosts:5
debug3: load_hostkeys_file: loaded 3 keys from mtkslt236
debug3: Failed to open file:C:/Users/mtk25836/.ssh/known_hosts2 error:2
debug1: load_hostkeys: fopen C:\\Users\\mtk25836/.ssh/known_hosts2: No such file or directory
debug3: Failed to open file:C:/ProgramData/ssh/ssh_known_hosts error:2
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts: No such file or directory
debug3: Failed to open file:C:/ProgramData/ssh/ssh_known_hosts2 error:2
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts2: No such file or directory
debug3: order_hostkeyalgs: have matching best-preference key type ssh-ed25519-cert-v01@openssh.com, using HostkeyAlgorithms verbatim
debug3: send packet: type 20
debug1: SSH2_MSG_KEXINIT sent
debug3: receive packet: type 20
debug1: SSH2_MSG_KEXINIT received
debug2: local client KEXINIT proposal
debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,ext-info-c
debug2: host key algorithms: ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,sk-ssh-ed25519@openssh.com,sk-ecdsa-sha2-nistp256@openssh.com,rsa-sha2-512,rsa-sha2-256,ssh-rsa
debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: compression ctos: none,zlib@openssh.com,zlib
debug2: compression stoc: none,zlib@openssh.com,zlib
debug2: languages ctos: 
debug2: languages stoc: 
debug2: first_kex_follows 0 
debug2: reserved 0 
debug2: peer server KEXINIT proposal
debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1
debug2: host key algorithms: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: compression ctos: none,zlib@openssh.com
debug2: compression stoc: none,zlib@openssh.com
debug2: languages ctos: 
debug2: languages stoc: 
debug2: first_kex_follows 0 
debug2: reserved 0 
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug3: send packet: type 30
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug3: receive packet: type 31
debug1: SSH2_MSG_KEX_ECDH_REPLY received
debug1: Server host key: ssh-ed25519 SHA256:lCbAZEo9ZlEwDts4qu6eVmdfdBDcQTE02xA46HPsILY
debug3: record_hostkey: found key type ECDSA in file C:\\Users\\mtk25836/.ssh/known_hosts:3
debug3: record_hostkey: found key type RSA in file C:\\Users\\mtk25836/.ssh/known_hosts:4
debug3: record_hostkey: found key type ED25519 in file C:\\Users\\mtk25836/.ssh/known_hosts:5
debug3: load_hostkeys_file: loaded 3 keys from mtkslt236
debug3: Failed to open file:C:/Users/mtk25836/.ssh/known_hosts2 error:2
debug1: load_hostkeys: fopen C:\\Users\\mtk25836/.ssh/known_hosts2: No such file or directory
debug3: Failed to open file:C:/ProgramData/ssh/ssh_known_hosts error:2
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts: No such file or directory
debug3: Failed to open file:C:/ProgramData/ssh/ssh_known_hosts2 error:2
debug1: load_hostkeys: fopen __PROGRAMDATA__\\ssh/ssh_known_hosts2: No such file or directory
debug1: Host 'mtkslt236' is known and matches the ED25519 host key.
debug1: Found key in C:\\Users\\mtk25836/.ssh/known_hosts:5
debug3: send packet: type 21
debug2: set_newkeys: mode 1
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug3: receive packet: type 21
debug1: SSH2_MSG_NEWKEYS received
debug2: set_newkeys: mode 0
debug1: rekey in after 134217728 blocks
debug3: unable to connect to pipe \\\\.\\pipe\\openssh-ssh-agent, error: 2
debug1: pubkey_prepare: ssh_get_authentication_socket: No such file or directory
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_rsa RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_dsa 
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ecdsa 
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ecdsa_sk 
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ed25519 
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_ed25519_sk 
debug1: Will attempt key: C:\\Users\\mtk25836/.ssh/id_xmss 
debug2: pubkey_prepare: done
debug3: send packet: type 5
debug3: receive packet: type 7
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug3: receive packet: type 6
debug2: service_accept: ssh-userauth
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug3: send packet: type 50
debug3: receive packet: type 51
debug1: Authentications that can continue: publickey,password
debug3: start over, passed a different list publickey,password
debug3: preferred publickey,keyboard-interactive,password
debug3: authmethod_lookup publickey
debug3: remaining preferred: keyboard-interactive,password
debug3: authmethod_is_enabled publickey
debug1: Next authentication method: publickey
debug1: Offering public key: C:\\Users\\mtk25836/.ssh/id_rsa RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 60
debug1: Server accepts key: C:\\Users\\mtk25836/.ssh/id_rsa RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs
debug3: sign_and_send_pubkey: RSA SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs
debug3: sign_and_send_pubkey: signing using rsa-sha2-512 SHA256:dL/RCjBMJP8GI+FP6ertWuA82eMBaXGmSzLKg4wfEDs
debug3: send packet: type 50
debug3: receive packet: type 52
debug1: Authentication succeeded (publickey).
Authenticated to mtkslt236 ([10.21.15.164]:22).
debug2: fd 5 setting O_NONBLOCK
debug2: fd 6 setting O_NONBLOCK
debug1: channel 0: new [client-session]
debug3: ssh_session2_open: channel_new: 0
debug2: channel 0: send open
debug3: send packet: type 90
debug1: Requesting no-more-sessions@openssh.com
debug3: send packet: type 80
debug1: Entering interactive session.
debug1: pledge: filesystem full
debug3: receive packet: type 80
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug3: client_input_hostkeys: received RSA key SHA256:mKyp8E5JRVKV+EN4hAZUrHJD4BWMdsz1PynJrZoDJFc
debug3: client_input_hostkeys: received ECDSA key SHA256:OEdwq8uaE3cCXLqzUgOMa2OqNzX+gYax4cxyD1ngZnk
debug3: client_input_hostkeys: received ED25519 key SHA256:lCbAZEo9ZlEwDts4qu6eVmdfdBDcQTE02xA46HPsILY
debug1: client_input_hostkeys: searching C:\\Users\\mtk25836/.ssh/known_hosts for mtkslt236 / (none)
debug3: hostkeys_foreach: reading file "C:\\Users\\mtk25836/.ssh/known_hosts"
debug2: C:\\Users\\mtk25836/.ssh/known_hosts:1: truncated before key type
debug3: hostkeys_find: found ecdsa-sha2-nistp256 key at C:\\Users\\mtk25836/.ssh/known_hosts:3
debug3: hostkeys_find: found ssh-rsa key at C:\\Users\\mtk25836/.ssh/known_hosts:4
debug3: hostkeys_find: found ssh-ed25519 key at C:\\Users\\mtk25836/.ssh/known_hosts:5
debug1: client_input_hostkeys: searching C:\\Users\\mtk25836/.ssh/known_hosts2 for mtkslt236 / (none)
debug3: Failed to open file:C:/Users/mtk25836/.ssh/known_hosts2 error:2
debug1: client_input_hostkeys: hostkeys file C:\\Users\\mtk25836/.ssh/known_hosts2 does not exist
debug3: client_input_hostkeys: 3 server keys: 0 new, 3 retained, 0 incomplete match. 0 to remove
debug1: client_input_hostkeys: no new or deprecated keys from server
debug3: receive packet: type 91
debug2: channel_input_open_confirmation: channel 0: callback start
debug2: fd 3 setting TCP_NODELAY
debug2: client_session2_setup: id 0
debug1: Sending command: 'exit'
debug2: channel 0: request exec confirm 1
debug3: send packet: type 98
debug2: channel_input_open_confirmation: channel 0: callback done
debug2: channel 0: open confirm rwindow 0 rmax 32768
debug2: channel 0: rcvd adjust 2097152
debug3: receive packet: type 99
debug2: channel_input_status_confirm: type 99 id 0
debug2: exec request accepted on channel 0
debug3: receive packet: type 96
debug2: channel 0: rcvd eof
debug2: channel 0: output open -> drain
debug2: channel 0: obuf empty
debug2: chan_shutdown_write: channel 0: (i0 o1 sock -1 wfd 5 efd 6 [write])
debug2: channel 0: output drain -> closed
debug3: receive packet: type 98
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
debug3: receive packet: type 98
debug1: client_input_channel_req: channel 0 rtype eow@openssh.com reply 0
debug2: channel 0: rcvd eow
debug2: chan_shutdown_read: channel 0: (i0 o3 sock -1 wfd 4 efd 6 [write])
debug2: channel 0: input open -> closed
debug3: receive packet: type 97
debug2: channel 0: rcvd close
debug3: channel 0: will not send data after close
debug2: channel 0: almost dead
debug2: channel 0: gc: notify user
debug2: channel 0: gc: user detached
debug2: channel 0: send close
debug3: send packet: type 97
debug2: channel 0: is dead
debug2: channel 0: garbage collecting
debug1: channel 0: free: client-session, nchannels 1
debug3: channel 0: status: The following connections are open:
  #0 client-session (t4 r0 i3/0 o3/0 e[write]/0 fd -1/-1/6 sock -1 cc -1)

debug3: send packet: type 1
debug3: fd 2 is not O_NONBLOCK
Transferred: sent 3212, received 2628 bytes, in 0.1 seconds
Bytes per second: sent 21629.2, received 17696.6
debug1: Exit status 0
close - IO is still pending on closed socket. read:1, write:0, io:000001BC14680E40
```

**Comment by hsinyinfu** [comment]
Oh, I see.
When I run the same command on my PC2 (no problem one), there is no the last few lines printed (shown in last comment):
```
debug1: channel 0: free: client-session, nchannels 1
debug3: channel 0: status: The following connections are open:
  #0 client-session (t4 r0 i3/0 o3/0 e[write]/0 fd -1/-1/6 sock -1 cc -1)

debug3: send packet: type 1
debug3: fd 2 is not O_NONBLOCK
Transferred: sent 3212, received 2628 bytes, in 0.1 seconds
Bytes per second: sent 21629.2, received 17696.6
debug1: Exit status 0
close - IO is still pending on closed socket. read:1, write:0, io:000001BC14680E40
```
But how does this happen? And how could I fix this problem?

**Comment by watson-github** [comment]
@hsinyinfu,  I suggest you try to use another terminal. I encounter the same issue as you. If I use the built-in OpenSSH in Visual Studio Code terminal on the same machine, there is no such issue.

**Comment by hsinyinfu** [comment]
@watson-github Could you please share the version of your built-in OpenSSH with me?
Also, you means that you encounter the same issue when you use the "Windows CMD" with OS built-in ssh, but the issue is not happened if you use the VS Code built-in terminal with the same OS built-in ssh on the same Windows machine?

**Comment by watson-github** [comment]
@hsinyinfu 
Yes, on the same machine, with the same built-in SSH, the same CommandPrompt. If I run CommadPrompt directly (native) or from [WindowsTerminal](https://github.com/Microsoft/Terminal), there is same issue as yours. And the shell could not exit until some delay elapsed after such "IO pending" message.
If I run from VS Code terminal, there is no "IO pending" message.

Unfortunately, I have updated my OpenSSH recently to solve another issue. I think it was an very old version, maybe OpenSSH_for_Windows_7. 

After I update OpenSSH to v8.9p1, there is still such "IO pending" message. But the shell can exit after showing this message. It is not so annoying now.

| | OpenSSH v7| OpenSSHv8.9p1|Other SSH (Cygwin or GitBash)
| --- | --- | --- | --- |
|WindowsTerminal |  IO pending message <br /> Exit with delay| IO pending message <br /> Exit without delay |No IO pending message <br /> Exit without delay|
| VSCode Terminal | No IO pending message <br /> Exit without delay | No IO pending message <br /> Exit without delay |No IO pending message <br /> Exit without delay|

It seems many factors are related with this issue. I have accepted this workaround. Just FYI, My Windows version is Version 1909(OS Build 18363.1440).
